### PR TITLE
Full Continuous Deployment CDEvents to render as CloudEvent and schema validation

### PR DESCRIPTION
**Pull request by adamkenihan**
This PR is a continuation of https://github.com/cdevents/sdk-java/pull/44, to implement other Continuous Deployment events

### CODE REVIEW

File: src/main/java/dev/cdevents/events/ServiceDeployedCDEvent.java [hunk 1]
```diff
@@ -0,0 +1,220 @@
...
+     */
+    public void setSubjectSource(URI subjectSource) {
+        getSubject().setSource(subjectSource);
+    }
+
+    /**
+     * @param subjectType
+     * sets the subject type
+     */
+    public void setSubjectType(CDEventConstants.SubjectType subjectType) {
```
Comment (rjalander) [inline src/main/java/dev/cdevents/events/ServiceDeployedCDEvent.java:193]: Subject Type will be set to "service" for this event, method to set the type is not needed.

File: src/main/java/dev/cdevents/events/ServicePublishedCDEvent.java [hunk 2]
```diff
@@ -0,0 +1,203 @@
...
+     */
+    public void setSubjectSource(URI subjectSource) {
+        getSubject().setSource(subjectSource);
+    }
+
+    /**
+     * @param subjectType
+     * sets the subject type
+     */
+    public void setSubjectType(CDEventConstants.SubjectType subjectType) {
```
Comment (rjalander) [inline src/main/java/dev/cdevents/events/ServicePublishedCDEvent.java:184]: same as above comment, can be removed.

File: src/main/java/dev/cdevents/events/ServiceRemovedCDEvent.java [hunk 3]
```diff
@@ -0,0 +1,203 @@
...
+     */
+    public void setSubjectSource(URI subjectSource) {
+        getSubject().setSource(subjectSource);
+    }
+
+    /**
+     * @param subjectType
+     * sets the subject type
+     */
+    public void setSubjectType(CDEventConstants.SubjectType subjectType) {
```
Comment (rjalander) [inline src/main/java/dev/cdevents/events/ServiceRemovedCDEvent.java:184]: same as above comment, can be removed.

File: src/main/java/dev/cdevents/events/ServiceRolledBackCDEvent.java [hunk 4]
```diff
@@ -0,0 +1,220 @@
...
+     */
+    public void setSubjectSource(URI subjectSource) {
+        getSubject().setSource(subjectSource);
+    }
+
+    /**
+     * @param subjectType
+     * sets the subject type
+     */
+    public void setSubjectType(CDEventConstants.SubjectType subjectType) {
```
Comment (rjalander) [inline src/main/java/dev/cdevents/events/ServiceRolledBackCDEvent.java:193]: same as above comment

File: src/main/java/dev/cdevents/events/ServiceUpgradedCDEvent.java [hunk 5]
```diff
@@ -0,0 +1,220 @@
...
+     */
+    public void setSubjectSource(URI subjectSource) {
+        getSubject().setSource(subjectSource);
+    }
+
+    /**
+     * @param subjectType
+     * sets the subject type
+     */
+    public void setSubjectType(CDEventConstants.SubjectType subjectType) {
```
Comment (rjalander) [inline src/main/java/dev/cdevents/events/ServiceUpgradedCDEvent.java:193]: same as above comment

**Review by rjalander — APPROVED** [review]

**Review by afrittoli — APPROVED** [review]

**Pull request merged**